### PR TITLE
Fix various gcc 14/g++ POSIX build issues

### LINKED ISSUE #22211: [MSWin] win32/GNUmakefile CFLAGS needs "-Wno-error=implicit-function-declaration" if building perl with gcc-14.1.0
**Description**

The mingw-w64 ports of gcc-14.1.0 currently available from https://winlibs.com turn "implicit declaration of function..." warnings into fatal errors.
This prevents ext/POSIX from building on all configurations.
It also stops -Duselongdouble builds at a very early stage because of an implicit function declaration (of <code>isnanl</code>) in sv.c.

Both of those failings can be addressed in win32/GNUmakefile by replacing:
```
EXTRACFLAGS	+= -std=c99
with
EXTRACFLAGS	+= -std=c99 -Wno-error=implicit-function-declaration
```
I'll make a PR for that in the next devel cycle, unless there's a better idea.

### PR DESCRIPTION
**Pull request by tonycoz**
Fixes #22211

### CODE REVIEW

File: ext/POSIX/POSIX.xs [hunk 1]
```diff
@@ -13,6 +13,23 @@
 
 static int not_here(const char *s);
 
+#if defined(__MINGW32__) && !defined(USE_QUADMATH)
+/* If nvtype is long double, the bessel functions still *
+ * operate at "double precision" only - as in the past. *
```
Comment (mauke) [inline ext/POSIX/POSIX.xs]: Why trailing `*` in these two lines?

Comment (tonycoz) [reply ext/POSIX/POSIX.xs]: probably word-wrapped it after some editing, I'll fix it

**Comment by sisyphus** [comment]
Thank you, @tonycoz .
This looks good and tests fine for me on both mingw-w64 and VS-2022 builds of perl-5.40.0-RC1.

If this can't be stuck into 5.40.0, what do we need to do to ensure it gets ported into 5.40.1 ?

File: ext/POSIX/POSIX.xs (hunk 1 above)
Comment (tonycoz) [reply ext/POSIX/POSIX.xs]: Fixed.

**Comment by jkeenan** [comment]
@tonycoz, @sisyphus:  Is this p.r. sufficiently well tested (particularly on Windows) so as to be ready to merge?

**Comment by sisyphus** [comment]
It will receive more extensive Windows testing in the building of Strawberry Perl-5.40.0.
So, even though I reckon it's right to go, it might be prudent to wait until SP-5.40.0 has been released.
My own testing has been reasonably extensive, but only using gcc-14.1.0 and only on Windows.

@shawnlaffan, am I correct in my beliefs that the SP-5.40.0 builds will:
1) include all of the patches contained in this PR;
2) be done using gcc-13.2.0.

I think we can afford wait for the Strawberry Perl release of 5.40.0  - unless it suffers a lengthy delay.

**Comment by shawnlaffan** [comment]
> am I correct in my beliefs that the SP-5.40.0 builds will:
>    1. include all of the patches contained in this PR;

  Yes, unless someone can provide reasons not to.

>    2. be done using gcc-13.2.0.

  Yes.

**Pull request merged**